qiskit/user_config.py

```diff
@@ -138,6 +139,72 @@ def read_config_file(self):
                 self.settings['num_processes'] = num_processes
 
 
+def set_config(key, value, section=None, file_path=None):
+    """Adds or modifies a user configuration
+
+    It will add the configuration in either the default location
+    ~/.qiskit/settings.conf or the value of file argument.
+
+    Only valid user config can be set in 'default' section. Custom
+    user config can be added in any other sections.
+
+    Args:
+        key (str): name of the config
+        value (str): value of the config
+        section (str, optional): if not specified, adds it to the
+            `default` section of the config file.
+        file_path (str, optional): the file to which config is added.
+            If not specified, adds it to the default config file or
+            if set, the value of `QISKIT_SETTINGS` env variable.
+
+    Raises:
+        QiskitUserConfigError: if the config is invalid
+    """
+    filename = file_path or os.getenv('QISKIT_SETTINGS', DEFAULT_FILENAME)
+    section = 'default' if section is None else section
+
+    if not isinstance(key, str):
+        raise exceptions.QiskitUserConfigError('Key must be string type')
+    if not isinstance(value, str):
+        raise exceptions.QiskitUserConfigError('Value must be string type')
```

Comment:
This will cause complaints since several of the values are ints or bools and it's annoying to have to convert those to strings up front. We should cast for people so you can do something like:

```python
set_config('transpiler_optimization_level', 3)
```